### PR TITLE
add rhsm to marshmallow

### PR DESCRIPTION
**Pull request by beav**
## Overview

This PR is being created to address [this Jira](https://issues.redhat.com/browse/HBISPSC-33).

This adds "rhsm" to match a similar PR in inventory schema.

## Secure Coding Practices Checklist GitHub Link

- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Review by tahmidefaz — CHANGES_REQUESTED** [review]
@beav we actually need the schema changes here as well. You can find the schema under `swagger/system_profile.spec.yaml`

**Review by tahmidefaz — APPROVED** [review]
LGTM :+1:

File: app/models.py [hunk 1]
```diff
@@ -327,6 +327,10 @@ class DiskDeviceSchema(MarshmallowSchema):
     type = fields.Str(validate=marshmallow_validate.Length(max=256))
 
 
+class RhsmSchema(MarshmallowSchema):
+    version = fields.Str(validate=marshmallow_validate.Length(max=256))
```
Comment (kruai) [inline app/models.py:331]: This is a tiny issue, but I think this should be `max=255` based on the change in the spec

**Review by kruai — CHANGES_REQUESTED** [review]

File: swagger/system_profile.spec.yaml [hunk 2]
```diff
@@ -324,3 +324,12 @@ $defs:
         maxLength: 10
         description: The SELinux mode provided in the config file
         example: 'permissive'
+      rhsm:
+        description: Object for subscription-manager details
+        type: object
+        properties:
+          version:
+            description: System release set by subscription-manager
+            example: "8.1"
```
Comment (ryandillinfelton) [inline swagger/system_profile.spec.yaml:333]: Is this example correct? If we're expecting strings like this 255 characters feels a little unnecessarily long.

Comment (kruai) [reply swagger/system_profile.spec.yaml:333]: That's also a good point

**Review by kruai — APPROVED** [review]
Looks like it matches up now. Did we decide to shorten the max length or keep it at 255?

**Comment by kruai** [comment]
> Looks like it matches up now. Did we decide to shorten the max length or keep it at 255?

I saw a conversation on another thread saying that the length is fine. LGTM :+1:

**Review by tahmidefaz — APPROVED** [review]
LGTM! :+1:

**Review by tahmidefaz — APPROVED** [review]
LGTM!

**Pull request merged**